### ISSUE TITLE
Downloads all of everything, including images, on first connection

### ISSUE THREAD

**Issue by ajorg-aws** [awslabs/pidgin-chime#27]
I seem to recall some discussion of this, but wanted to be sure it was logged as an issue. When you first log in, all of your conversations begin to download, seemingly from the beginning of time. This includes any images you were sent in that time. It can take a very long time to download, and your computer is very busy for the duration.

**Comment by ajorg-aws** [comment]
Images are entered into the conversation as they arrive, and not at the location in the history where they would have been, so they pile up at the bottom.

**Comment by dwmw2** [comment]
Yeah, there is no way for us to "silently" insert old messages into the logs, other than playing them all as new messages when we first connect. It would be a whole lot nicer to do that differently. We *do* now have a notification when a given conversation is opened, which we use for sending read receipts. Maybe we could use that, and fetch the messages for a given conversation only when it's first opened? I think we already do manage that for chat rooms, don't we?

Ordering of images clearly *is* a bug. We should play those messages in order. I wonder if we could abuse the same trick that the chimeseen plugin does for displaying read receipts, finding the message in the window and "amending" it.

Ideally this gets fixed in Pidgin upstream though, to allow this stuff to work without hacks.
